internal/service/price.go

```diff
@@ -650,27 +652,92 @@ func (s *priceService) GetPrices(ctx context.Context, filter *types.PriceFilter)
 }
 
 func (s *priceService) UpdatePrice(ctx context.Context, id string, req dto.UpdatePriceRequest) (*dto.PriceResponse, error) {
-	price, err := s.PriceRepo.Get(ctx, id)
-	if err != nil {
+	// Validate the request
+	if err := req.Validate(); err != nil {
 		return nil, err
 	}
 
-	price.Description = req.Description
-	price.Metadata = req.Metadata
-	price.LookupKey = req.LookupKey
-
-	if err := s.PriceRepo.Update(ctx, price); err != nil {
+	// Get the existing price
+	existingPrice, err := s.PriceRepo.Get(ctx, id)
+	if err != nil {
 		return nil, err
 	}
 
-	response := &dto.PriceResponse{Price: price}
+	// Check if the request has critical fields
+	if req.HasCriticalFields() {
+		if existingPrice.EndDate != nil {
+			return nil, ierr.NewError("price is already terminated").
+				WithHint("Cannot update a terminated price").
+				WithReportableDetails(map[string]interface{}{
+					"price_id": id,
+				}).
+				Mark(ierr.ErrValidation)
+		}
 
-	// TODO: !REMOVE after migration
-	if price.EntityType == types.PRICE_ENTITY_TYPE_PLAN {
-		response.PlanID = price.EntityID
-	}
+		// Set termination end date - use EndDate from request if provided, otherwise use current time
+		terminationEndDate := time.Now().UTC()
+		if req.EndDate != nil {
+			terminationEndDate = *req.EndDate
+		}
 
-	return response, nil
+		// Terminate the existing price
+		existingPrice.EndDate = &terminationEndDate
```

Comment:
make this critical only and expect the required amount related fields based on right billing model so that this is actually update only